### ISSUE TITLE
[HEALTH 7.0.1] await health.requestAuthorization(healthTypes) returning true on Android after denying pop-up twice

### ISSUE THREAD

**Issue by Kam5678** [cph-cachet/flutter-plugins#779]
I am trying to figure out a solution to the use case where the user denies permissions to health connect twice on the following screen:

![unnamed (1)](https://github.com/cph-cachet/flutter-plugins/assets/44074596/5296bc2e-758f-4c98-adba-7bb3339d016f)

Then I viewed that health.requestAuthorization(healthTypes) will return false both times **which is expected.**

But after that, lets say : 

### health.requestAuthorization(healthtypes) runs a third time, the pop-up will not appear and in the console log it will say access is granted to HealthConnect which is not the case.

Since I use this boolean value to determine if HealthConnect has allowed access to my app, it is essential this boolean value returns the correct value otherwise **I am faced with the issue of not knowing when I am allowed to fetch healthConnect data.**

Or if possible is there an alternative way to check HealthConnect permissions for my current app on flutter?
(Did some reserach but couldn't find a way, other than the current **health.requestAuthorization** function being used in this package)

###  Device Used: Galaxy Tab A, Android Version 11 API 33

Flutter Doctor:

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.10.5, on Microsoft Windows [Version 10.0.19045.3208], locale en-CA)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.5)  
[√] Android Studio (version 2022.2)
[√] VS Code (version 1.79.1)
[√] Connected device (4 available)
[√] Network resources

• No issues found!
```

**Comment by ICP-Theo** [comment]
I'am facing the same problem. 

debug output:
I/FLUTTER_HEALTH( 9739): Access Granted (to Health Connect)!
.
.
W/FLUTTER_HEALTH::ERROR( 9739): [Health Connect] There was an error adding the workout
W/FLUTTER_HEALTH::ERROR( 9739): MNCBicycle lacks the following permissions: [android.permission.health.WRITE_EXERCISE, android.permission.health.WRITE_DISTANCE, android.permission.health.WRITE_TOTAL_CALORIES_BURNED]

requestAuthorization returns true.
This is a serious error

**Comment by itsamk** [comment]
Same issue!

**Comment by hoffmatteo** [comment]
Hello, thank you for your observations!

I was able to reproduce your error and it will be fixed in the upcoming version of health. After a third time of requesting the permissions, Android simply blocks the request but still returns it as a success, just with an added note. This is why health returned it as true, even though that is incorrect.

In terms of the popup not appearing I am not sure what exactly the behaviour is, since this is done by Android itself. It could be that there is some timer until it can appear again, but so far the only way to consistently get it back was for me to reinstall both HC and the app itself.

Additionally, if you want to verify the permissions again, this can be done using hasPermissions.

**Comment by hoffmatteo** [comment]
Fixed in latest release.

**Comment by Kam5678** [comment]
Yeah I agree, I believe after denying pop-up twice, android simply doesn't show the pop-up again, and you basically have to manually set it in the settings page.

Thank you for looking into the permissions fixes!
